### PR TITLE
Add from parameter instead of report type

### PR DESCRIPTION
**Pull request by aadityataparia**
- Takes from parameter (date from which it will fetch prs)
- Remove report type parameter
  - In line with report different reports and including them to one class

- There are a few half-assed changes, but they will be filled by later PRs :'( 

- And removed methods unrelated to PRs fetching - will be moving these to new base report branch, so that each report class only need to take prs as initialize parameter

### CODE REVIEW

**Comment by aadityataparia** [comment]
I know the concerns that's why it is hard to explain with one PR, as there are not many independant changes that can be devided into PRs, 🙆‍♂️ that's I wanted to have a meeting with you so that I could explain the working and it would have been easier to merge the changes

**Comment by aadityataparia** [comment]
But I will reduce this to only changing from parameter! But next PR will be either too big, or too hard to explain :'(

**Comment by aadityataparia** [comment]
> I imagine that for each type of report, we will probably need different kind of parameters, so maybe we can find a better approach than hardcoding this into the base report?

Not really, So I have changed report classes to take prs as argument only, so whatever PRs we give they will generate report for it. So if we want one week/ 10day/ 1 month report, we can specify that with from parameter 🤔

**Comment by aadityataparia** [comment]
@balvig ♻️

File: lib/mergometer.rb [hunk 1]
```diff
@@ -3,19 +3,18 @@
 
 require "mergometer/version"
 require "mergometer/configuration"
-
-reports_path = File.expand_path("../mergometer/reports/**/*.rb", __FILE__)
-Dir[reports_path].each { |f| require f }
```
Comment (balvig) [inline lib/mergometer.rb]: (This change seems unrelated? Maybe not necessary? `Report` itself does not depend on the individual reports)

File: lib/mergometer/report.rb [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+Dir[reports_path].each { |f| require f }
+
 module Mergometer
   class Report
     require "hirb"
 
-    def initialize(repo)
+    def initialize(repo, from = nil)
       @repo = repo
+      @from = from || (Date.today - 1.day).beginning_of_week.strftime("%Y-%m-%d")
```
Comment (balvig) [inline lib/mergometer/report.rb]: This is a little hard for me to parse 🙇  when is this? Yesterday, but the beginning of the week?

File: lib/mergometer/report.rb [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 
+reports_path = File.expand_path("./reports/**/*.rb", __FILE__)
+Dir[reports_path].each { |f| require f }
+
 module Mergometer
   class Report
     require "hirb"
 
-    def initialize(repo)
+    def initialize(repo, from = nil)
```
Comment (balvig) [inline lib/mergometer/report.rb]: I still feel like something like `(repo, **options={})` would be better here, and then allowing each sub report to define what kind of options it needs. Some reports might benefit from specifying `from: ...`, others might now?

File: lib/mergometer/report.rb (hunk 2 above)
Comment (aadityataparia) [reply lib/mergometer/report.rb]: Yeah, I thought if we run script on sunday, it should get prs from last week rather than only for sunday 🤔

File: lib/mergometer.rb (hunk 1 above)
Comment (aadityataparia) [reply lib/mergometer.rb]: Yeah I guess (it does later though :P). Will remove this.

File: lib/mergometer/report.rb (hunk 3 above)
Comment (aadityataparia) [reply lib/mergometer/report.rb]: https://github.com/balvig/mergometer/pull/2#issuecomment-383803809

But I guess you are right, using option is more flexible!

**Comment by aadityataparia** [comment]
Current use -> bin/run #{repo} #{report_type} #{from}

**Comment by aadityataparia** [comment]
FYI: I will open more PRs after coming back from holidays (7th May).

**Comment by aadityataparia** [comment]
Moving to #6
